gtsam/precompiled_header.h

```diff
@@ -42,8 +42,10 @@
 #include <gtsam/base/Matrix.h>
 #include <gtsam/base/OptionalJacobian.h>
 #include <gtsam/base/ProductLieGroup.h>
+#ifdef GTSAM_ENABLE_BOOST_SERIALIZATION
```

Comment:
This is an awesome catch. However, can you please put this macro guard within `serialization.h`, similar to `serializationTestHelpers.h`?

The latter has all the code within the `#ifdef GTSAM_ENABLE_BOOST_SERIALIZATION` guard. Doing this for `serialization.h` will ensure other such issues are avoided.